### PR TITLE
typescript-node: clean up require and import

### PR DESCRIPTION
**Pull request by laubfall**
… instead of import statement

- Resolves https://github.com/OpenAPITools/openapi-generator/issues/4656
- Fixed broken link in CONTRIBUTING.md

### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) beforehand.
- [x] Run the shell script `./bin/generate-samples.sh`to update all Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. These must match the expectations made by your contribution. You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/config/java*`. For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

@TiFu (2017/07) @taxpon (2017/07) @sebastianhaas (2017/07) @kenisteward (2017/07) @Vrolijkx (2017/09) @macjohnny (2018/01) @topce (2018/10) @akehir (2019/07) @petejohansonxo (2019/11) @amakhrov (2020/02)

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/typescript-node/api-single.mustache [hunk 1]
```diff
@@ -1,6 +1,13 @@
 {{>licenseInfo}}
+
+{{#supportsES6}}
+import localVarRequest from 'request';
+import http from 'http';
+{{/supportsES6}}
+{{^supportsES6}}
 import localVarRequest = require('request');
```
Comment (DanielRuf) [inline modules/openapi-generator/src/main/resources/typescript-node/api-single.mustache:8]: The import var = require does not make much sense to me. This should be probably `const localVarRequest = require('request')`.

Or did I oversee something?

Comment (DanielRuf) [reply modules/openapi-generator/src/main/resources/typescript-node/api-single.mustache:8]: Same for the other parts.

**Comment by DanielRuf** [comment]
> * [ x]

You might want to remove the space inside the square brackets (it is either a space or x as single character) ;-)

**Comment by macjohnny** [comment]
@laubfall could you please fix the CI issues?

File: modules/openapi-generator/src/main/resources/typescript-node/api-single.mustache (hunk 1 above)
Comment (laubfall) [reply modules/openapi-generator/src/main/resources/typescript-node/api-single.mustache:8]: Yeah thats right, i'll fix it :)

Comment (laubfall) [reply modules/openapi-generator/src/main/resources/typescript-node/api-single.mustache]: I'm getting some typescript errors with const http = require. These are "TS2503: Cannot find namespace 'http'." and "error TS4055: Return type of public method from exported class has or is using private name 'http'.". As described here: https://stackoverflow.com/questions/31173738/typescript-getting-error-ts2304-cannot-find-name-require there are some ways to make this work, but to be honest i don't see a reason why to make a difference at this point. All other modules are imported with the "import .. from" stmt independent of the ES6 flag. So i removed the switch and changed the tsconfig to make the imports http/request work. What do you think about this solution?

Comment (DanielRuf) [reply modules/openapi-generator/src/main/resources/typescript-node/api-single.mustache]: For TypeScript to use these you need the types for the Nodejs internal ones. Basically `@types/node`.

Comment (DanielRuf) [reply modules/openapi-generator/src/main/resources/typescript-node/api-single.mustache]: The current solution should be fine.

**Comment by wing328** [comment]
Thanks for the PR but your commit (as shown in the Commits tab) is not linked to your Github account, which means this PR won't count as your contribution in https://github.com/OpenAPITools/openapi-generator/graphs/contributors.

Let me know if you need help fixing it.

Ref: https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-can-i-update-commits-that-are-not-linked-to-my-github-account

**Comment by wing328** [comment]
I just pushed https://github.com/OpenAPITools/openapi-generator/pull/6947/commits/547e30521f96f8879fae76eca197ab7df20b99cc to fix the CI failure. Let's see how that goes.

File: samples/client/petstore/typescript-node/default/.openapi-generator/FILES [hunk 2]
```diff
@@ -5,10 +5,10 @@ api/petApi.ts
 api/storeApi.ts
 api/userApi.ts
 git_push.sh
-model/apiResponse.ts
-model/category.ts
-model/order.ts
-model/pet.ts
-model/tag.ts
-model/user.ts
+model/./apiResponse.ts
```
Comment (DanielRuf) [inline samples/client/petstore/typescript-node/default/.openapi-generator/FILES]: why this change again?

Comment (DanielRuf) [inline samples/client/petstore/typescript-node/default/.openapi-generator/FILES]: Why this change? ./ makes not much sense in my opinion.

Comment (wing328) [reply samples/client/petstore/typescript-node/default/.openapi-generator/FILES]: Agreed `./` is not needed. We'll tackle it separately.

Comment (laubfall) [reply samples/client/petstore/typescript-node/default/.openapi-generator/FILES]: Ok, so there is nothing left here to todo for the PR?

**Comment by laubfall** [comment]
> I just pushed [547e305](https://github.com/OpenAPITools/openapi-generator/commit/547e30521f96f8879fae76eca197ab7df20b99cc) to fix the CI failure. Let's see how that goes.

Thanks, i looked into the CI Failure but did not see the reason why the build failed

**Comment by wing328** [comment]
Shippable CI failure not related to this PR.
